### ISSUE TITLE
Separator between items

### ISSUE THREAD

**Issue by CRMemory** [newsboat/newsboat#412]
Version: newsboat r2.13

I'd like to have a separator (e.g. a line) inbetween feeds for an easy grouped overview. In the `urls` file, this could be done for example by a line starting with `---` where everything after the dashes is ignored (so that a section label can be given there)

**Comment by zaowen** [comment]
Thanks @CRMemory for taking the time to make this issue.
I had some questions, how were you imagining the `---` would look like in the feed view? At present three dashes can be put into the `urls` file and it will still show up in the view as an empty feed, visually separating your feeds. 
Also, did you know about newsboat's [tagging](https://newsboat.org/releases/2.14/docs/newsboat.html#_tagging) feature? It sounds similiar to what you might be looking for.

**Comment by CRMemory** [comment]
@zaowen Thanks for getting back.
A visual separation is exactly what I had in mind. I had imagined a completely blank line, without an index number or the number of entries. I want to get a quick overview which "sections" have unread items in, when glancing over it.
It's somewhat similar to tagging, but the tagging overview ("t") doesn't show which tag contains new elements. (Or is there a way for that)
Your proposal with three dashes as empty feed works for now (sorry, haven't tried that before), just doesn't look as tidy. Consider the issue a low-priority enhancement wish.

**Comment by doneforaiur** [comment]
Bump. An option to put three dashes between feeds when they are `feed-sort-order firsttag`ged would be great.

**Comment by Minoru** [comment]
@doneforaiur As far as I understand, the idea here was to group feeds together like this:

```
--- First group
https://example.com/news1.atom "one"
https://example.com/news2.rss "two"
--- Second group
https://example.com/news3.atom "one"
https://example.com/news4.atom "two"
https://example.com/news5.atom "one"
```

That only works as long as `feed-sort-order none` is used, i.e. feeds are shown in the order they're listed in the urls file. If any other sorting is used, the groups are broken up, which IMHO makes them useless. @doneforaiur, given the above urls file, can you show how the feedlist should look under `feed-sort-order firsttag`? Or am I completely misunderstanding your use-case?

**Comment by CRMemory** [comment]
I think @doneforaiur and my proposals are similar but different: My original proposal had `feed-sort-order none` in mind (i.e. separators as listed in the urls file. The issue is that sorting by `firsttag` also sorts the tags alphabetically but I have a logic of the feeds/tags in the urls file.

Taking the urls file as @Minoru has shown: I guess with the `feed-sort-order none` the feed list should the feeds in order of the urls file with separations where indicated in the file (`---` was the proposal here)
```
--- First group
https://example.com/news1.atom "one"
https://example.com/news2.rss "two"
--- Second group
https://example.com/news3.atom "one"
https://example.com/news4.atom "two"
https://example.com/news5.atom "one"
```

whereas with `feed-sort-order firsttag` the feed list should have separators according to the given tags which is how I understand @doneforaiur 
```
--- "One"
https://example.com/news1.atom "one"
https://example.com/news3.atom "one"
https://example.com/news5.atom "one"
--- "Two"
https://example.com/news2.rss "two"
https://example.com/news4.atom "two"
```

**Comment by doneforaiur** [comment]
@CRMemory, yes, that's exactly how I imagined it. Altough as @Minoru said, it makes `firsttag` obselete. It would be nice to have an indicator when `feed-sort-order firsttag` is set. It's a minor improvement but the fact that `---` is clickable and counts as a feed seems unnecesary, as @CRMemory also mentioned in their second post.

**Comment by Minoru** [comment]
@doneforaiur Got it. Given we haven't implemented the "simple" separators yet, I've no idea when we'll get around to the fancier ones that you're describing. But please do file an issue about them, mentioning this one. (I want a separate issue because this one is about "simple" separators, and it will be closed once those are implemented).

> It would be nice to have an indicator when feed-sort-order firsttag is set.

Can you file a separate issue describing your use-case where indication is better than just typing `gf`? Sorting is fast, I don't see why you wouldn't just sort the feedlist again when you are unsure about its current ordering.

> the fact that --- is clickable and counts as a feed seems unnecesary

I think this will be fixed automatically once we teach feedlist to display not just feeds but "separators" as well.

**Comment by A1RO** [comment]
Even though it's not exactly straightforward, you can add blank lines and colorful section headers to your feedlist.
Here's how I just configured mine, which is quite similar to what @CRMemory asked.

1. In the `urls` file, I create a blank line (notice the whitespace betwen quotes) followed by a section header / separator:

```
" "
"--- Section Name ---"
https://example.com/feed tag
```

2. In the `config` file, I highlight the header, then blank lines with empty feeds (keeping only the header or feed title):

```
highlight feedlist "---.*---" green black bold
highlight feedlist "^.*\\(0/0\\)" black black
```

Have a look at [Highlight Text](https://newsboat.org/releases/2.20.1/docs/newsboat.html#_highlighting_text) for details.

3. When you combine this technique with the tagging feature mentioned by @zaowen, you get a neat overview of all the feeds as a homepage, along with a filtered view when needed.

And thanks @Minoru for your work here! I thought I could share a bit to help on this journey.

**Comment by jfrech** [comment]
As a comment to @A1RO's suggestion: When working with a terminal with a non-black default background, it may be prettier to set
```
highlight feedlist "^.*\\(0/0\\)" default default invis
```
to not introduce black-colored background artifacts.

**Comment by A1RO** [comment]
Following @jfrech comment, I changed all of my [color](https://newsboat.org/releases/2.21/docs/newsboat.html#_configuring_colors) and [highlight](https://newsboat.org/releases/2.21/docs/newsboat.html#_highlighting_text) backgrounds to `default`, plus added the `invis` attribute as mentioned. Even with a dark background, the `black` color was lighter than my terminal's background. Of course, results may vary with different color schemes.

**Comment by reportaman** [comment]
+1 this feature would be great to have

**Comment by ignamartinoli** [comment]
+1 would love it

**Comment by chromer030** [comment]
+1

**Comment by camoz** [comment]
I'm also interested in this. Personally, I only have 20-30 feeds, and it's very easy for me to categorize them (i.e. 1:1 tag relation). Having a straightforward way to categorize them in the config file would be neat. For me, separator lines would be sufficient for that, ideally with custom text in the separator (= the category name), but that's not a must, as I only have 4 categories and I can remember those :) That would be a bit more convenient than tagging every feed, but more importantly it would visually set them apart in newsboat.

**Comment by nsklaus** [comment]
why not use '#' commented line, on top of a bunch of feed links ?
would work the same as requested "---" but it's "#" instead
like:
```
# feeds about history
https://some-link.net/blah/rss       "history" ~some_site
https://another-link.net/feeds/rss  "history" ~site_name
https://other-link.net/news/rss      "history" ~my_history_site
# feeds about carpentry
link1                                                 "carpentry" ~name1 
link2                                                "carpentry" ~name2
link3                                                "carpentry" ~name3
# feeds about sport
..

```

**Comment by camoz** [comment]
@nsklaus AIUI this issue is about a visual separator in newsboat's feed view, not in its config file.

**Comment by nsklaus** [comment]
that was for "urls" file, not "config" but i understand what you meant.
if that's the case then my comment is off target, sorry about that :)
